pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/codestyle/xml/UnnecessaryCast.xml

```diff
@@ -154,4 +154,449 @@ public class MapCasts {
 }
         ]]></code>
     </test-code>
+    <test-code>
+        <description>Unnecessary (primitive -> primitive) casts</description>
+        <expected-problems>4</expected-problems>
+        <expected-linenumbers>5,6,7,8</expected-linenumbers>
+        <code><![CDATA[
+            class Scratch {
+                public static void main(String[] args) {
+                    Integer integer = 2; // ok
+
+                    long a = (int) Integer.valueOf(2);  // explicit boxing where the value would be autoboxed
+                    int b = (int) 2;                    // unnecessary, it's the same type
+                    long c = (int) 4;                   // widening can be implicit
+                    double c = (int) 4;                 // widening can be implicit
+                    double c = (int) 4.0;               // cast is narrowing so necessary
+                }
+            }
+            ]]></code>
+    </test-code>
+    <test-code>
+        <description>Tests from the doc</description>
+        <expected-problems>2</expected-problems>
+        <code><![CDATA[
+            class Scratch {
+                public static void main(String[] args) {
+                    Object context = (Comparable) "o";
+                    Integer integer = (Integer) 1;
+                }
+            }
+            ]]></code>
+    </test-code>
+    <test-code>
+        <description>Context is array index</description>
+        <expected-problems>2</expected-problems>
+        <expected-linenumbers>3,4</expected-linenumbers>
+        <code><![CDATA[
+            class Scratch {
+                public static void main(String[] args, Integer i, Double d) {
+                    args[(int) i] = "";   // unnecessary
+                    args[(int) '0'] = ""; // unnecessary
+                    args[(int) 0.5] = ""; // narrowing so necessary
+                    args[(int) d] = "";   // unboxing + narrowing so necessary
+                }
+            }
+            ]]></code>
+    </test-code>
+    <test-code>
+        <description>Context is invocation, no overloads</description>
+        <!-- TODO for now this is explicitly disabled -->
```

Comment:
Maybe use here `regressionTest="false"`? Then you could set the expected-problems correct already

Reply:
Actually no, because we want to be sure that there is no violation here. If we use the attribute, then the test wouldn't be run, and there may or may not be a violation